src/engine/Sim3D.ts

```diff
@@ -358,32 +358,44 @@ export class Sim3D {
 
     if (worldConfig.walls) {
       // We want walls
-      if (worldConfig.walls.length === 0) {
-        // // Empty array, generate the default set of walls (perimeter)
+      if (!worldConfig.walls.walls) {
```

Comment:
Let's leave the `worldConfig.walls` check as it was, but add a `console.warn` call in it that states it will be deprecated soon.

Right before the `if (worldConfig.walls)` check, add a check for `if (worldConfig.perimeter)` so that we can favor the `perimeter` property over walls. This should also simplify the logic a little bit.